[Elton Chua] iP

notle1706 opened the pull request:
## JamieBot ##
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Jamie makes your already busy routine more convenient by remembering things you need to do. It's
- text-based
- very fast
- ~easy to learn~ SUPER EASY TO LEARN

All you need to do is to
1. download from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you. :hand_over_mouth:

Get it **FREE** for a _limited_ time only!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

notle1706 committed:
Changed Duke.java to Jamie.java

notle1706 committed:
Changed file name back to Duke
Level 2 done

notle1706 committed:
Added Tasks.java
Editted Duke.java

notle1706 committed:
Added Todo.java
Added Event.java
Added Deadline.java

notle1706 committed:
Added some exceptions

notle1706 committed:
Added deleteTask() in Task.java
Added a few more exceptions

notle1706 committed:
Added DukeException class
Did Automated Text UI Testing
Added a few java docs

notle1706 committed:
Update conditionals style

notle1706 committed:
Refactor Duke to Jamie and fix bugs

notle1706 committed:
Add JamieTime.java

notle1706 committed:
Add OOP files

notle1706 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

notle1706 committed:
Add save to text function

notle1706 committed:
Add Time and Date

notle1706 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Jamie.java
#	src/main/java/Task.java

notle1706 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Event.java
#	src/main/java/Parser.java
#	src/main/java/Task.java

notle1706 committed:
Merge branch 'master' of https://github.com/notle1706/ip

# Conflicts:
#	build.gradle
#	gradlew
#	src/main/java/Event.java
#	src/main/java/Jamie.java
#	src/main/java/Task.java
#	src/main/java/Ui.java

notle1706 committed:
Add OOP

notle1706 committed:
Add packages

notle1706 committed:
Add Junit tests

notle1706 committed:
Package into JAR file

notle1706 committed:
Add Java Doc

notle1706 committed:
Improve coding standards

notle1706 committed:
Add Find feature

notle1706 committed:
Merge branch 'branch-A-JavaDoc'

notle1706 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/ToDo.java

notle1706 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/ToDo.java

notle1706 committed:
Add final to Deadline and Event class variable

notle1706 committed:
Merge branch 'branch-A-CodingStandard'

notle1706 committed:
Standardise jamie to duke

notle1706 committed:
Add checkstyle

rama-pang commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public abstract class Task {
+    protected String item;
+    protected boolean done;
```
Perhaps this should be:
```suggestion
    protected boolean isDone;
```
instead? According to [Java coding standards](https://se-education.org/guides/conventions/java/intermediate.html), boolean variables/methods should be named to sound like booleans

notle1706 replied:
Thank you for pointing it out. I've made the changes.

rama-pang commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,55 @@
+     */
+    public abstract String toLine();
+
+    /**
+     * Sets the task as done.
+     */
+    public void setDone() {
+        this.done = true;
+    }
+
+    /**
+     * Sets the task as undone.
+     */
+    public void setUnDone() {
+        this.done = false;
+    }
```
Perhaps this could be compacted to:
```suggestion
    public void setDone(boolean isDone) {
        this.isDone = isDone;
    }
```
instead? According to [Java coding standards](https://se-education.org/guides/conventions/java/intermediate.html), setter methods for boolean variables must be of the form:
```
void setFound(boolean isFound);
```

notle1706 replied:
Thank you for catching this.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+    /**
+     * Prints a welcome message when the program starts.
+     */
+    public void welcome() {
```
Perhaps this should be:
```suggestion
    public void showWelcome() {
```
instead? According to [Java coding standards](https://se-education.org/guides/conventions/java/intermediate.html), names representing methods must be verbs.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * Prints out the message when a task is added.
+     * @param task The task that is added.
+     */
+    public void addTasksUi(Task task) {
```
Perhaps the method name could be renamed to something similar to `printAddTaskMessage`, as the name `addTasksUi` could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * @param marking The marking that is being done.
+     * @param task The task that is being marked.
+     */
+    public void markingUi(String marking, Task task) {
```
Perhaps the method name could be renamed to something similar to `printMarkTaskMessage`, as the name `markingUi`  is not a verb and could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * Prints out the message for when a task is deleted.
+     * @param task The task that is deleted.
+     */
+    public void deleteUi(Task task) {
```
Perhaps the method name could be renamed to something similar to `printDeleteTaskMessage`, as the name `deleteUi` could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * Prints out the message that displays the number of tasks in the list of tasks.
+     * @param tasks The list of tasks.
+     */
+    public void numberOfTasksUi(TaskList tasks) {
```
Perhaps the method name could be renamed to something similar to `printNumberOfTasks`, as the name `numberOfTasksUi` is not a verb and could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * Prints out an error message when there are missing numbers.
+     * @param command The command that was called.
+     */
+    public void missingNumberAfterCommand(String command) {
```
Perhaps the method name could be renamed to something similar to `printMissingNumberAfterCommandError`, as the name `missingNumberAfterCommand` is not a verb and could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+    /**
+     * Prints out an error message when an invalid command is called.
+     */
+    public void invalidCommand() {
```
Perhaps the method name could be renamed to something similar to `printInvalidCommandError`, as the name `invalidCommand` is not a verb and could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * Prints out an error message when something is missing in the description of a task.
+     * @param task The task that has a missing description.
+     */
+    public void someThingMissingError(String task) {
```
Perhaps the method name could be renamed to something similar to `printSomeThingMissingError`, as the name `someThingMissingError` is not a verb and could be misleading.

rama-pang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+    /**
+     * The error message for when there is no such task.
+     */
+    public void noSuchTaskError() {
```
Perhaps the method name could be renamed to something similar to `printNoSuchTaskError`, as the name `noSuchTaskError` is not a verb and could be misleading.

rama-pang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,87 @@
+                        String[] taskInfo = line.split("[*]", 4);
+
+                        switch (taskInfo[0]) {
+                            case "T":
```
According to [Java Coding Standards](https://se-education.org/guides/conventions/java/intermediate.html), there should be no indentation for `case` clauses.

notle1706 committed:
Add GUI

notle1706 committed:
Merge branch 'branch-A-CheckStyle'

notle1706 committed:
Merge branch 'branch-Level-10'

notle1706 committed:
Improve JavaDoc

notle1706 committed:
Fix indentation for switch statement

ketamethane commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,100 @@
+                byInput = input.substring(end + 4);
+            } catch (StringIndexOutOfBoundsException e) {
+                return duke.ui.someThingMissingError("deadline");
+            }
+            Deadline deadline = new Deadline(textInput, byInput);
+            duke.tasks.addTask(deadline);
+            return duke.ui.addTasksUi(deadline) + duke.ui.numberOfTasksUi(duke.tasks);
+        } else if (input.startsWith("event ")) {
+            int end = input.indexOf("/at ");
+            String textInput;
+            String atInput;
+            try {
+                textInput = input.substring(6, end);
+                atInput = input.substring(end + 4);
+            } catch (StringIndexOutOfBoundsException e) {
+                return duke.ui.someThingMissingError("event");
+            }
+            Event event = new Event(textInput, atInput);
+            duke.tasks.addTask(event);
+            return duke.ui.addTasksUi(event) + duke.ui.numberOfTasksUi(duke.tasks);
+        } else if (input.startsWith("delete ")) {
+            try {
+                int number = Integer.parseInt(input.substring(7));
+                Task removed = duke.tasks.deleteTask(number);
+                return duke.ui.deleteUi(removed) + duke.ui.numberOfTasksUi(duke.tasks);
+            } catch (DukeException e) {
+                return duke.ui.noSuchTaskError();
+            } catch (StringIndexOutOfBoundsException e) {
+                return duke.ui.missingNumberAfterCommand("delete");
+            }
+        } else if (input.startsWith("find ")) {
+            try {
+                String keyWord = input.substring(5);
+                TaskList tempList = duke.tasks.find(keyWord);
+                return duke.ui.printTaskList(tempList);
+            } catch (DukeException e) {
+                return duke.ui.noSuchTaskError();
+            }
+        } else {
+            return duke.ui.invalidCommand();
```
Perhaps you can consider using switch-cases instead?

ketamethane commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,108 @@
+                + "\ndd/MM/yyyy HHmm eg. 29/10/2022 0000";
+    }
+
+    /**
+     * Returns the taskList in string form.
+     * @param tasks The taskList that is being printed.
+     * @return The tasks in the taskList in string form.
+     */
```
I think there should be a line break after the description (line 18). The same goes for all the other JavaDoc comments.

notle1706 replied:
Thank you for letting me know about this.

ketamethane reviewed (commented):
Hello! I was tasked with reviewing your PR coding quality (particularly naming). Overall, I am quite impressed with the clear naming. Even the methods are explicitly named and conveyed to me what to expect without reading the JavaDoc. 
I made the same mistake of not having a line break after the description in the JavaDoc as well, until my reviewer pointed this out to me.

notle1706 committed:
Add assert statements

notle1706 committed:
Combine setter methods for isDone

There are two methods for setting isDone.

Combining the two methods makes the code in line with CS2103T
coding standards.

notle1706 committed:
Merge pull request #1 from notle1706/branch-A-Assertions

Add assert statements

notle1706 committed:
Merge branch 'master' into branch-A-CodeQuality

notle1706 committed:
Merge pull request #2 from notle1706/branch-A-CodeQuality

Improve coding standards

notle1706 committed:
Merge branch 'master' of https://github.com/notle1706/ip

notle1706 committed:
Add FixedDurationTask.java

notle1706 committed:
Merge branch 'branch-B-FixedDurationTasks'

notle1706 committed:
Add mass operations feature

notle1706 committed:
Merge branch 'branch-C-MassOps'

notle1706 committed:
Update Readme.md

notle1706 committed:
Update README.md

notle1706 committed:
Add Ui.png
